src/genegraph/transform/jsonld/common.clj

```diff
@@ -72,37 +73,62 @@
-       (log/trace :msg "Setting jsonld options")
-                                        ; TODO does nothing
-                                        ;(.setOmitGraph jsonld-options true)
-                                        ; TODO does nothing
-                                        ;(.setProcessingMode jsonld-options "JSON_LD_1_1")
-       (.set context JsonLDWriter/JSONLD_OPTIONS jsonld-options)
-       (log/trace :msg "Writing framed jsonld")
-       (.write writer sw ds prefix-map base-uri context)
-       (.toString sw)))))
+      (comment
+        (let [writer (JsonLDWriter. RDFFormat/JSONLD_COMPACT_PRETTY)
+              sw (StringWriter.)
+              ds (DatasetGraphInMemory.)
+           ; prefix-map left blank
+              prefix-map (PrefixMapStd.)
+              base-uri ""
+              context (Context.)
+              jsonld-options (JsonLdOptions.)]
+          (.setUseNativeTypes jsonld-options true)
+       ;(.setCompactArrays jsonld-options false)
+          (log/trace :msg "Adding model to dataset")
+       ; we don't use the graph name on export, its value shouldn't appear in output
+          (.addGraph ds (NodeFactory/createURI "BLANK") (.getGraph model))
+          (log/trace :msg "Setting jsonld frame")
+       ; TODO this frame option appears to do nothing when writing jsonld
+       ; maybe it affects reading, not sure why it's under JsonLDWriter then though
+       ;(.set context JsonLDWriter/JSONLD_FRAME frame-str)
+          (log/trace :msg "Setting jsonld options")
+       ; TODO does nothing
+       ;(.setOmitGraph jsonld-options true)
+       ; TODO does nothing
+       ;(.setProcessingMode jsonld-options "JSON_LD_1_1")
+          (.set context JsonLDWriter/JSONLD_OPTIONS jsonld-options)
+          (log/trace :msg "Writing framed jsonld")
+          (.write writer sw ds prefix-map base-uri context)
+          (.toString sw))))))
+
+(defn model-to-jsonld [^Model model]
+  (let [sw (doto (StringWriter.)
+             (RDFDataMgr/write model RDFFormat/JSONLD_COMPACT_PRETTY))]
+    (.toString sw)))
+
+(comment
+  ;; There is a frame field on the Jena Context object, but it is not used
+  ;; in the implementation of JsonLD11Writer, which is the class used by RDFWriter (an RDFDataMgr)
+
+  (defn frame-jsonld-4-5 [^String input ^String frame-str]
+    (log/info :fn :frame-jsonld-4-5 :input input :frame-str frame-str)
+    (let [model ^Model (l/read-rdf (str->bytestream input) {:format :json-ld})
+          write-ctx (JsonLDWriteContext.)]
+      (.setFrame write-ctx frame-str)
+      (let [rdf-writer (-> (doto (RDFWriter/create)
+                           ;; TODO JSONLD11_FRAME_PRETTY
+                             (.format RDFFormat/JSONLD10_FRAME_PRETTY)
+                             (.source model)
+                             (.context write-ctx))
+                           .build)]
+        (let [sw (StringWriter.)]
+          (.output rdf-writer sw)
+          (.toString sw))))))
```

Comment:
This comment I will leave in because it shows how to implement the JSONLD 1.1 framing when that is incorporated into the core Jena libraries. Which can then obsolete our `jsonld-to-jsonld-framed` function.